core/src/main/java/com/yahoo/oak/OakHashMap.java

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import java.util.AbstractCollection;
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * A concurrent map implementation which supports off-heap memory.
+ */
+public class OakHashMap<K, V>  extends AbstractMap<K, V> implements AutoCloseable, ConcurrentZCMap<K , V> {
+
+    /*
+     * Memory manager cares for allocation, de-allocation and reuse of the internally pre-allocated
+     * memory. There can be separate memory managing algorithms for keys and values.
+     * */
+    private final MemoryManager valuesMemoryManager;
+    private final MemoryManager keysMemoryManager;
+    private final OakTransformer<K> keyDeserializeTransformer;
+    private final OakTransformer<V> valueDeserializeTransformer;
+    private final Function<Map.Entry<OakScopedReadBuffer, OakScopedReadBuffer>,
+            Map.Entry<K, V>> entryDeserializeTransformer;
+    private final OakComparator<K> comparator;
+    private final InternalOakMap<K , V> internalOakMap;
+
+
+
+    private final K fromKey = null; //TODO: to be removed when internalOakHashMap  implementation is done
+    private final boolean fromInclusive = false; //TODO: to be removed when internalOakHashMap  implementation is done
+    private final K toKey = null; //TODO: to be removed when internalOakHashMap  implementation is done
+    private boolean toInclusive = false; //TODO: to be removed when internalOakHashMap  implementation is done
+    private final boolean isDescending = false; //TODO: to be removed when internalOakHashMap  implementation is done
```

Comment:
We don't have to defer the removal of these for later.
Simply correct the relevant code to not use these values.